plugins/doc_fragments/consul.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) Ansible project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+class ModuleDocFragment(object):
+    # Common parameters for Consul modules
+    DOCUMENTATION = r"""
+options:
+  host:
+    description:
+      - Host of the consul agent, defaults to V(localhost).
+    default: localhost
+    type: str
+  port:
+    type: int
+    description:
+      - The port on which the consul agent is running.
+    default: 8500
+  scheme:
+    description:
+      - The protocol scheme on which the consul agent is running.
+    default: http
+    choices: [ http, https ]
```

Comment:
Adding the `choices` here is a breaking change. (Or was it absolutely impossible to specify anything else than `http` and `https` as the scheme before?)

Reply:
I wouldn't know what else one could specify. But I'll happily remove it if you think it is to dangerous.

Reply:
I guess it's better to not have choices here then. You could mention both choices in the `description` if you want, if you want to make sure users know what they can specify here (that the second choices is `https` is probably not completely obvious to everyone).

Reply:
It is a weird setting anyways, in the long run it would probably make more sense to deprecate this and have a simple ssl on/off toggle.

Reply:
If this change remains, this PR cannot be merged because it contains a breaking change.